deviations/models.py

```diff
@@ -86,6 +86,19 @@ class SubmissionRuleDeviation(UrlMixin, models.Model):
         verbose_name=_('LABEL_SUBMITTER'),
         on_delete=models.CASCADE,
     )
+    granter = models.ForeignKey(UserProfile,
+        verbose_name=_('LABEL_GRANTER'),
+        on_delete=models.CASCADE,
```

Comment:
`on_delete` should actually be SET_NULL, right? If the teacher user account is deleted, the student's deviation should still exist. This field even already has `null=True` and `blank=True`.

https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.SET_NULL

If you agree, you can change it here and also in the migration. Don't generate two separate migrations!

Reply:
I agree, fixed.